plutus-core/cost-model/create-cost-model/CreateBuiltinCostModel.hs

```diff
@@ -235,241 +228,217 @@ createBuiltinCostModel bmfile rfile = do
+    "added_sizes"          -> ModelTwoArgumentsAddedSizes         <$> readOneVariableLinearFunction "I(x_mem + y_mem)" e
+    "subtracted_sizes"     -> ModelTwoArgumentsSubtractedSizes    <$> error "subtracted sizes"
+    "multiplied_sizes"     -> ModelTwoArgumentsMultipliedSizes    <$> readOneVariableLinearFunction "I(x_mem * y_mem)" e
+    "min_size"             -> ModelTwoArgumentsMinSize            <$> readOneVariableLinearFunction "pmin(x_mem, y_mem)" e
+    "max_size"             -> ModelTwoArgumentsMaxSize            <$> readOneVariableLinearFunction "pmax(x_mem, y_mem)" e
+    "linear_on_diagonal"   -> ModelTwoArgumentsLinearOnDiagonal   <$> readTwoVariableFunConstOrLinear "x_mem" e
+    "const_below_diagonal" -> ModelTwoArgumentsConstBelowDiagonal <$> readTwoVariableFunConstOr loadTwoVariableCostingFunction e
+    "const_above_diagonal" -> ModelTwoArgumentsConstAboveDiagonal <$> readTwoVariableFunConstOr loadTwoVariableCostingFunction e
+    "quadratic_in_y"       -> ModelTwoArgumentsQuadraticInY       <$> readOneVariableQuadraticFunction "y_mem" e
+    _                      -> error $ "Unknown two-variable model type: " ++ ty
+
+loadThreeVariableCostingFunction :: MonadR m => SomeSEXP (Region m) -> m ModelThreeArguments
+loadThreeVariableCostingFunction e = do
+  ty <- getType e
+  case ty of
+    "constant_cost"               -> ModelThreeArgumentsConstantCost          <$> getConstant e
+    "linear_in_x"                 -> ModelThreeArgumentsLinearInX             <$> readOneVariableLinearFunction "x_mem" e
+    "linear_in_y"                 -> ModelThreeArgumentsLinearInY             <$> readOneVariableLinearFunction "y_mem" e
+    "linear_in_z"                 -> ModelThreeArgumentsLinearInZ             <$> readOneVariableLinearFunction "z_mem" e
+    "quadratic_in_z"              -> ModelThreeArgumentsQuadraticInZ          <$> readOneVariableQuadraticFunction "z_mem" e
+    "literal_in_y_or_linear_in_z" -> ModelThreeArgumentsLiteralInYOrLinearInZ <$> error "literal"
+    _                             -> error $ "Unknown three-variable model type: " ++ ty
+
+loadSixVariableCostingFunction :: MonadR m => SomeSEXP (Region m) -> m ModelSixArguments
+loadSixVariableCostingFunction e = do
+  ty <- getType e
+  case ty of
+    "constant_cost" -> ModelSixArgumentsConstantCost <$> getConstant e
+    _               -> error $ "Unknown six-variable model type: " ++ ty
 
 boolMemModel :: ModelTwoArguments
 boolMemModel = ModelTwoArgumentsConstantCost 1
 
 memoryUsageAsCostingInteger :: ExMemoryUsage a => a -> CostingInteger
 memoryUsageAsCostingInteger = coerce . sumCostStream . flattenCostRose . memoryUsage
 
+-- | The types of functions which take an R SEXP and extract a CPU model from
+-- it, then pair it up with a memory costing function.
+type MakeCostingFun1 = forall m . MonadR m => SomeSEXP (Region m) -> m (CostingFun ModelOneArgument)
```

Comment:
Have a single

```
type MakeCostingFun model = forall m . MonadR m => SomeSEXP (Region m) -> m (CostingFun model)
```

instead and use it as `MakeCostingFun ModelTwoArguments`? Even if only for defining the same type synonyms.

Reply:
I thought there would probably be a more generic way of doing this but I was in a hurry!  I'll think about that again later.

Reply:
I think doing it generically isn't gonna be any better, my version of `MakeCostingFun` merely eliminates some of the boilerplate.